src/Component/src/Metadata/Metadata.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata;
+
+use Doctrine\Inflector\Inflector as InflectorObject;
+use Doctrine\Inflector\InflectorFactory;
+
+final class Metadata implements MetadataInterface
+{
+    private string $name;
+
+    private string $applicationName;
+
+    /** @var string */
+    private $driver;
+
+    private ?string $stateMachineComponent = null;
+
+    /** @var string */
+    private $templatesNamespace;
+
+    /** @var array */
+    private $parameters;
+
+    private static ?InflectorObject $inflectorInstance = null;
+
+    private function __construct(string $name, string $applicationName, array $parameters)
+    {
+        $this->name = $name;
+        $this->applicationName = $applicationName;
+
+        $this->driver = $parameters['driver'];
+        $this->templatesNamespace = array_key_exists('templates', $parameters) ? $parameters['templates'] : null;
+        $this->stateMachineComponent = $parameters['state_machine_component'] ?? null;
+
+        $this->parameters = $parameters;
+    }
+
+    public static function fromAliasAndConfiguration(string $alias, array $parameters): self
```

Comment:
This method missing in interface, but used in registry.

```php
    public function addFromAliasAndConfiguration(string $alias, array $configuration): void
    {
        $this->add(Metadata::fromAliasAndConfiguration($alias, $configuration));
    }
```

Should it be moved intro external factory? Otherwise it's kind of defeats the reason to have `MetadataInterface`